src/Sql/Sql/Elastic Jobs/Cmdlet/JobStep/GetAzureSqlElasticJobStep.cs

```diff
@@ -0,0 +1,243 @@
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.Sql.ElasticJobs.Model;
+
+namespace Microsoft.Azure.Commands.Sql.ElasticJobs.Cmdlet.Job
+{
+    /// <summary>
+    /// Defines the Get-AzureRmSqlElasticJobStep Cmdlet
+    /// </summary>
+    [Cmdlet(VerbsCommon.Get, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "SqlElasticJobStep",
+        DefaultParameterSetName = DefaultParameterSet)]
+    [OutputType(typeof(AzureSqlElasticJobStepModel))]
+    public class GetAzureSqlElasticJobStep : AzureSqlElasticJobStepCmdletBase<AzureSqlElasticJobModel>
+    {
+        /// <summary>
+        /// The parameter sets
+        /// </summary>
+        protected const string DefaultGetVersionParameterSet = "GetVersion";
+        protected const string ParentObjectGetVersionParameterSet = "GetVersionUsingJobObject";
+        protected const string ParentResourceIdGetVersionParameterSet = "GetVersionUsingParentResourceId";
+
+        /// <summary>
+        /// Gets or sets the resource group name
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = DefaultParameterSet,
+            Position = 0,
+            HelpMessage = "The resource group name")]
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = DefaultGetVersionParameterSet,
+            Position = 0,
+            HelpMessage = "The resource group name")]
+        public override string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the job input object
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = InputObjectParameterSet,
+            ValueFromPipeline = true,
+            Position = 0,
+            HelpMessage = "The job input object")]
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = ParentObjectGetVersionParameterSet,
+            ValueFromPipeline = true,
+            Position = 0,
+            HelpMessage = "The job input object")]
+        [ValidateNotNullOrEmpty]
+        public AzureSqlElasticJobModel ParentObject { get; set; }
+
+        /// <summary>
+        /// Gets or sets the job resource id
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = ResourceIdParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            Position = 0,
+            HelpMessage = "The job resource id")]
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = ParentResourceIdGetVersionParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            Position = 0,
+            HelpMessage = "The job resource id")]
+        [ValidateNotNullOrEmpty]
+        public string ParentResourceId { get; set; }
```

Comment:
@johnpaulkee same comment about `-ResourceId`

Reply:
As discussed we'll address the Resource Id changes in a separate PR to reduce complexity of this one. We can do a one-off release with these changes and then add -ResourceId parameter  later.